### PR TITLE
update to latest (v0.0.32) of natscli

### PR DESCRIPTION
**Pull request by Houlistonm**
update to latest (v0.0.32) of natscli, some jq improvements; some filters for json

### CODE REVIEW

File: http-routes/040-route-propagation-pt-2.markdown [hunk 1]
```diff
@@ -59,7 +59,7 @@ to.
    actualLRPs. Cfdot is a helpful CLI for using the BBS API.  It's a great tool
    for debugging on the Diego Cell.
  ```
- cfdot actual-lrps | jq .
+ cfdot actual-lrps | jq -re .
```
Comment (ameowlia) [inline http-routes/040-route-propagation-pt-2.markdown]: Why do you want to add the `-re` flag here?

File: http-routes/050-route-propagation-pt-3.markdown [hunk 2]
```diff
@@ -48,17 +48,17 @@ technique.
...
 0. Get the NATS cli
     ```
-wget https://github.com/nats-io/natscli/releases/download/0.0.25/nats-0.0.25-linux-amd64.zip
-unzip nats-0.0.25-linux-amd64.zip
-chmod +x nats-0.0.25-linux-amd64
-mv nats-0.0.25-linux-amd64/nats /usr/bin
+wget https://github.com/nats-io/natscli/releases/download/v0.0.32/nats-0.0.32-linux-amd64.zip
+unzip nats-0.0.32-linux-amd64.zip
+chmod +x nats-0.0.32-linux-amd64
+mv nats-0.0.32-linux-amd64/nats /usr/bin
```
Comment (ameowlia) [inline http-routes/050-route-propagation-pt-3.markdown:54]: 👍 This part looks good.

File: http-routes/050-route-propagation-pt-3.markdown [hunk 3]
```diff
@@ -48,17 +48,17 @@ technique.
...
 
     ```
 0. Get NATS username, password, and server address
     ```
-    cat /var/vcap/jobs/route_emitter/config/route_emitter.json | jq . | grep nats
+    jq -re '[to_entries[] | select(.key | startswith("nats")) ] | from_entries' /var/vcap/jobs/route_emitter/config/route_emitter.json
```
Comment (ameowlia) [inline http-routes/050-route-propagation-pt-3.markdown]: I think this change sacrifices understanding for efficiency. I would rather keep the old command, which I find more readable.

File: http-routes/050-route-propagation-pt-3.markdown [hunk 4]
```diff
@@ -48,17 +48,17 @@ technique.
...
     ```
-0. Use the nats cli to connect to nats: `nats sub "*.*" -s nats://NATS_USERNAME:NATS_PASSWORD@NATS_ADDRESS`. The `"*.*"` means that you are subscribing to all NATs messages.
+0. Use the nats cli to connect to nats: `nats sub "router.register" -s nats://NATS_USERNAME:NATS_PASSWORD@NATS_ADDRESS/ --tlscert <cert file from json> --tlskey <key file from json> --tlsca <ca file from json>`. Using `"*.*"` means that you are subscribing to all NATs messages.
```
Comment (ameowlia) [inline http-routes/050-route-propagation-pt-3.markdown]: 1. Re changing "*.*" to "router.register"
I purposefully wanted readers to subscribe to all nats messages to see how many there are and how many things use nats. I'd be open to another step after this one that shows the reader how to subscribe to just the router messages.

2. Re adding the cert information
👍  thanks!

File: http-routes/060-route-propagation-pt-4.markdown [hunk 5]
```diff
@@ -44,7 +44,8 @@ Let's take a look at that route table.
  ```
 0. Get the routes table
  ```
- curl "http://USERNAME:PASSWORD@localhost:8080/routes" | jq .
+ curl -s "http://USERNAME:PASSWORD@localhost:8080/routes" | jq -re '[ to_entries[] | select(.value[].tags.app_name == "appA") ] | from_entries' # return just appA
```
Comment (ameowlia) [inline http-routes/060-route-propagation-pt-4.markdown]: Again I think this change sacrifices understanding for efficiency. I would rather keep the old command, which I find more readable.

Also the next step asks the user to scroll though and look at all the routes and all the information associated with them, so I don't want jq pruning out any information.

File: http-routes/080-incoming-http-requests-pt-1.markdown [hunk 6]
```diff
@@ -40,7 +40,8 @@ listening at on the Diego Cell and specifically at
 1. Ssh onto the Diego Cell where appA is deployed and become root.
 1. Use netstat to look at open ports
  ```
- netstat -tulp
+ netstat -ntulp
```
Comment (ameowlia) [inline http-routes/080-incoming-http-requests-pt-1.markdown:43]: 👍

**Review by ameowlia — CHANGES_REQUESTED** [review]
🎉 Thank you so much for opening _the very first PR on this project_. 

I can see from your changes that you are a jq super user. I am hesitant to change many of the commands you suggested. I prefer making the commands as readable and understandable as possible, even if they do lack efficiency. 

However, I think jq is a powerful tool that is useful to learn.  If you are interested, I am open to you adding another story to this track or maybe a bonus section at the bottom of some of the stories with tips on how to do things with jq.

File: http-routes/040-route-propagation-pt-2.markdown (hunk 1 above)
Comment (Houlistonm) [reply http-routes/040-route-propagation-pt-2.markdown]: Honestly, it more muscle memory, the -r to unquote, and the -e to report return codes.
Just to get people into the habit of typing the command out.

File: http-routes/050-route-propagation-pt-3.markdown (hunk 3 above)
Comment (Houlistonm) [reply http-routes/050-route-propagation-pt-3.markdown]: Just trying to sneak some jq into peoples lives. :)

File: http-routes/050-route-propagation-pt-3.markdown [hunk 7]
```diff
@@ -82,6 +82,7 @@ mv nats-0.0.25-linux-amd64/nats /usr/bin
      }
 }
  ```
+   Nats is used by multiple services, and the `router.register` messages are intermixed with other messages.  Modify the sub filter from `"*.*"` to `"router.register"` to view only router messages.
```
Comment (ameowlia) [inline http-routes/050-route-propagation-pt-3.markdown:85]: Love it!

**Review by ameowlia — APPROVED** [review]

**Comment by ameowlia** [comment]
🎉  First PR merged!!! 

Thank you so much for these contributions @Houlistonm!

**Pull request merged**